Examples/ObjectiveC/LineGradientExample.m

```diff
@@ -0,0 +1,58 @@
+#import "LineGradientExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleLineGradient = @"LineGradientExample";
+
+@interface LineGradientExample ()<MGLMapViewDelegate>
+
+@end
+
+@implementation LineGradientExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle lightStyleURL]];
+
+    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    
+    mapView.delegate = self;
+    
+    // Set the map’s center coordinate and zoom level.
+    [mapView setCenterCoordinate:CLLocationCoordinate2DMake(38.875, -77.035)
+                       zoomLevel:12
+                        animated:NO];
+    
+    [self.view addSubview:mapView];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    
+    NSURL *url = [NSURL fileURLWithPath:[[NSBundle mainBundle] pathForResource:@"line-gradient" ofType:@"geojson"]];
+    MGLShapeSource *source = [[MGLShapeSource alloc] initWithIdentifier:@"line-source" URL:url options:nil];
```

Comment:
Note that the clarification that will happen in https://github.com/mapbox/mapbox-gl-native/issues/13000 should allow you to switch back to using an `MGLPolyline` like you did in 100c88d.